File: leetcode_ly/java/String/RomanToInteger.java
public class Solution {
    public int romanToInt(String s) {
        if(s.isEmpty()) return 0;
        char cur = s.charAt(0);;
        int i = 0;
        int res = 0;
        while(i < s.length()){
            switch(s.charAt(i)){
                case 'I':
                    res += 1;
                    break;
                case 'V':
                    if(cur == 'I') res += 3;
                    else res += 5;
                    break;
                case 'X':
                    if(cur == 'I') res += 8;
                    else res += 10;
                    break;
                case 'L':
                    if(cur == 'X') res += 30;
                    else res += 50;
                    break;
                case 'C':
                    if(cur == 'X') res += 80;
                    else res += 100;
                    break;
                case 'D':
                    if(cur == 'C') res += 300;
                    else res += 500;
                    break;
                case 'M':
                    if(cur == 'C') res += 800;
                    else res += 1000;
                    break;
                default:
                    break;
            }
            cur = s.charAt(i);
            i++;
        }
        return res;
    }
}

File: leetcode_ly/c++/easy/RomanToInteger.cpp
class Solution {
public:
    int romanToInt(string s) {
        char cur = s[0];
        int i = 0;
        int result = 0;
        while(i < s.size()){
            switch(s[i]){
                case 'I':
                    result += 1;
                    break;
                case 'V':{
                    if(cur == 'I')
                        result += 3;
                    else
                        result += 5;
                    break;
                }
                case 'X':{
                    if(cur == 'I')
                        result += 8;
                    else
                        result += 10;
                    break;
                }
                case 'L':{
                    if(cur == 'X')
                        result += 30;
                    else
                        result += 50;
                    break;
                }
                case 'C':{
                    if(cur == 'X')
                        result += 80;
                    else
                        result += 100;
                    break;
                }
                case 'D':{
                    if(cur == 'C')
                        result += 300;
                    else
                        result += 500;
                    break;
                }
                case 'M':{
                    if(cur == 'C')
                        result += 800;
                    else
                        result += 1000;
                    break;
                }
                default:
                    break;
            }
            cur = s[i];
            i++;
        }
        return result;
    }
};